pypika/terms.py

```diff
@@ -385,9 +439,55 @@ def get_formatted_value(cls, value: Any, **kwargs):
             return "null"
         return str(value)
 
-    def get_sql(self, quote_char: Optional[str] = None, secondary_quote_char: str = "'", **kwargs: Any) -> str:
-        sql = self.get_value_sql(quote_char=quote_char, secondary_quote_char=secondary_quote_char, **kwargs)
-        return format_alias_sql(sql, self.alias, quote_char=quote_char, **kwargs)
+    def get_sql(
+        self,
+        quote_char: Optional[str] = None,
+        secondary_quote_char: str = "'",
+        parameter: Parameter = None,
+        **kwargs: Any,
+    ) -> str:
+        if parameter is None:
+            sql = self.get_value_sql(quote_char=quote_char, secondary_quote_char=secondary_quote_char, **kwargs)
+            return format_alias_sql(sql, self.alias, quote_char=quote_char, **kwargs)
+        else:
+            # Don't stringify numbers when using a parameter
+            if isinstance(self.value, (int, float)):
+                value_sql = self.value
+            else:
+                value_sql = self.get_value_sql(quote_char=quote_char, **kwargs)
+            param_sql = parameter.get_sql(**kwargs)
+            param_key = parameter.get_param_key(placeholder=param_sql)
+            parameter.update_parameters(param_key=param_key, value=value_sql, **kwargs)
+
+            return format_alias_sql(param_sql, self.alias, quote_char=quote_char, **kwargs)
+
+
+class ParameterValueWrapper(ValueWrapper):
+    def __init__(self, parameter: Parameter, value: Any, alias: Optional[str] = None) -> None:
+        super().__init__(value, alias)
+        self._parameter = parameter
+
+    def get_sql(
+        self,
+        quote_char: Optional[str] = None,
+        secondary_quote_char: str = "'",
+        parameter: Parameter = None,
+        **kwargs: Any,
+    ) -> str:
+        if parameter is None:
+            sql = self.get_value_sql(quote_char=quote_char, secondary_quote_char=secondary_quote_char, **kwargs)
+            return format_alias_sql(sql, self.alias, quote_char=quote_char, **kwargs)
+        else:
+            # Don't stringify numbers when using a parameter
+            if isinstance(self.value, (int, float)):
+                value_sql = self.value
+            else:
+                value_sql = self.get_value_sql(quote_char=quote_char, **kwargs)
+            param_sql = self._parameter.get_sql(**kwargs)
+            param_key = self._parameter.get_param_key(placeholder=param_sql)
+            parameter.update_parameters(param_key=param_key, value=value_sql, **kwargs)
+
+            return format_alias_sql(param_sql, self.alias, quote_char=quote_char, **kwargs)
```

Comment:
This seems to be the same as the `get_sql` above. Could it be inherited somehow? Maybe wrapped up in a common function?

Reply:
There is a slight difference, But I've broken it out into `_get_param_data`